test/src/graph/nodes/RaytraceNodeTest.cpp

```diff
@@ -21,3 +29,200 @@ TEST_F(RaytraceNodeTest, valid_arguments)
+}
+
+TEST_F(RaytraceNodeTest, config_velocity_invalid_argument_linear_velocity)
+{
+	ASSERT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr));
+
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_node_raytrace_configure_velocity(raytraceNode, nullptr, nullptr),
+	                            "linear_velocity != nullptr");
+}
+
+TEST_F(RaytraceNodeTest, config_velocity_invalid_argument_angular_velocity)
+{
+	ASSERT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr));
+
+	const rgl_vec3f linearVelocity = {1.0f, 2.0f, 3.0f};
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_node_raytrace_configure_velocity(raytraceNode, &linearVelocity, nullptr),
+	                            "angular_velocity != nullptr");
+}
+
+TEST_F(RaytraceNodeTest, config_velocity_invalid_node_object)
+{
+	const rgl_vec3f linearVelocity = {1.0f, 2.0f, 3.0f};
+	const rgl_vec3f angularVelocity = {4.0f, 5.0f, 6.0f};
+	EXPECT_RGL_INVALID_OBJECT(rgl_node_raytrace_configure_velocity((rgl_node_t) 0x1234, &linearVelocity, &angularVelocity),
+	                          "Object does not exist: Node 0x1234");
+}
+
+TEST_F(RaytraceNodeTest, config_velocity_valid_arguments)
+{
+	ASSERT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr));
+
+	const rgl_vec3f linearVelocity = {1.0f, 2.0f, 3.0f};
+	const rgl_vec3f angularVelocity = {4.0f, 5.0f, 6.0f};
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace_configure_velocity(raytraceNode, &linearVelocity, &angularVelocity));
+}
+
+TEST_F(RaytraceNodeTest, config_distortion_invalid_argument_node)
+{
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_node_raytrace_configure_distortion(raytraceNode, false), "node != nullptr");
+}
+
+TEST_F(RaytraceNodeTest, config_distortion_invalid_node_object)
+{
+	EXPECT_RGL_INVALID_OBJECT(rgl_node_raytrace_configure_distortion((rgl_node_t) 0x1234, false),
+	                          "Object does not exist: Node 0x1234");
+}
+
+TEST_F(RaytraceNodeTest, config_distortion_valid_arguments)
+{
+	ASSERT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr));
+
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace_configure_distortion(raytraceNode, true));
+}
+
+TEST_F(RaytraceNodeTest, config_non_hit_distance_invalid_argument_node)
+{
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_node_raytrace_configure_non_hit_distance_values(raytraceNode, 0.0f, 0.0f),
+	                            "node != nullptr");
+}
+
+TEST_F(RaytraceNodeTest, config_non_hit_distance_invalid_node_object)
+{
+	EXPECT_RGL_INVALID_OBJECT(rgl_node_raytrace_configure_non_hit_distance_values((rgl_node_t) 0x1234, 0.0f, 0.0f),
+	                          "Object does not exist: Node 0x1234");
+}
+
+TEST_F(RaytraceNodeTest, config_non_hit_distance_valid_arguments)
+{
+	ASSERT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr));
+
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace_configure_non_hit_distance_values(raytraceNode, 0.0f, 0.0f));
+}
+
+TEST_F(RaytraceNodeTest, config_non_hit_near_distance_should_correctly_change_output)
```

Comment:
I'd set up the scene for both hits and non-hits to be produced. The test will validate whether hits are not changed.
The same goes for the second test.